invocation.md

```diff
@@ -0,0 +1,1170 @@
+
+type Selector union {
+  | Key     String
+  | Path    [String]
+} kinded
+
+type InvocationReference union {
+  | Invocation<Any>
+  | &Invocation<Any>
+} representation kinded
+```
+
+# 3 Invocation
+
+An invocation is the smallest unit of work that can be requested from a UCAN. It describes one `(resource, ability, input)` triple. The `input` field is free form, and depend on the specific resource and ability being interacted with, and not described in this specification.
+
+Using the JavaScript analogy from the introduction, an invocation is similar to function application:
+
+```json
+{
+  "v": "0.1.0",
+  "iss": "did:key:z6Mkqa4oY9Z5Pf5tUcjLHLUsDjKwMC95HGXdE1j22jkbhz6r",
+  "aud": "did:web:ucan.run",
+
+  "with": "mailto:alice@example.com",
+  "do": "msg/send",
+  "inputs": {
+    "to": ["bob@example.com", "carol@example.com"],
+    "subject": "hello",
+    "body": "world"
+  },
+
+  "prf": [
+    { "/": "bafkreie2cyfsaqv5jjy2gadr7mmupmearkvcg7llybfdd7b6fvzzmhazuy" }
+  ],
+  "s": {
+    "/": {
+      "bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+    }
+  }
+}
+```
+
+```js
+// Pseudocode
+msg.send("mailto:alice@example.com", {
+  to: ["bob@example.com", "carol@example.com"],
+  subject: "hello",
+  body: "world",
+})
+```
+
+Later, when we explore [Promise]s, this also includes capturing the promise:
+
+```json
+{
+  "v": "0.1.0",
+  "iss": "did:key:z6Mkqa4oY9Z5Pf5tUcjLHLUsDjKwMC95HGXdE1j22jkbhz6r",
+  "aud": "did:web:ucan.run",
+  "with": "mailto://alice@example.com",
+  "do": "msg/send",
+  "input": {
+    "to": {
+      "<-": [
+        {
+          "v": "0.1.0",
+          "iss": "did:key:zAlice",
+          "aud": "did:web:ucan.run",
+          "with": "https://example.com/mailinglist",
+          "do": "crud/read",
+          "s": {
+            "/": {
+              "bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+            }
+          },
+          "prf": [{"/": "bafybeia3tspzaay4gcx3npcczgidbrsutq7yxnag3sfzrmvua6ogqjwy7a"}]
+        },
+        "ok"
+      ],
+    }
+    "subject": "hello",
+    "body": "world"
+  },
+  "prf": [
+    { "/": "bafkreie2cyfsaqv5jjy2gadr7mmupmearkvcg7llybfdd7b6fvzzmhazuy" }
+  ],
+  "s": {
+    "/": {
+      "bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+    }
+  }
+}
+```
+
+```js
+// Pseudocode
+const mailingList = crud.read("https://exmaple.com/mailinglist", {})
+msg.send("mailto:alice@example.com", {
+  to: mailingList.catch(error => error).then(result => result.ok),
+  subject: "hello",
+  body: "world",
+})
+```
+
+## 3.1 Schema
+
+```ipldsch
+type Invocation<In> struct {
+  v      SemVer
+
+  iss    Principal
+  aud    Principal
+
+  with   URI
+  do     Ability
+  input  In (implicit {})
+
+  meta   {String : Any} (implicit {})
+
+  prf    [&UCAN]
+  nnc    optional String
+  nbf    optional Int
+
+  s      Varsig
```

Comment:
I think I'm contractually obligated to note that IPVM can't align on forcing a signature here

Reply:
After our talk I have created this PR to capture what I thought we agreed on https://github.com/web3-storage/specs/pull/35/files

I have not updated this PR as I was hoping we could align on remaining pieces before I go ahead and update everything else.